types/fabric/fabric-impl.d.ts

```diff
@@ -4480,3 +4481,50 @@ interface IUtil extends IUtilAnimation, IUtilArc, IObservable<IUtil>, IUtilDomEv
 	object: IUtilObject;
 	string: IUtilString;
 }
+
+export interface FilterBackend {
+	resources: Object;
+
+	applyFilters(filters: IBaseFilter[], sourceElement: HTMLImageElement | HTMLCanvasElement, sourceWidth: number, sourceHeight: number, targetCanvas: HTMLCanvasElement): any;
+
+	evictCachesForKey(cacheKey: any): void;
+
+	dispose(): void;
+
+	clearWebGLCaches(): void;
+
+}
+export let filterBackend: FilterBackend;
+export interface Canvas2dFilterBackend extends FilterBackend { }
+export class Canvas2dFilterBackend {
+	constructor();
+}
+
+export interface WebglFilterBackendOptions {
+	tileSize: number;
+}
+export interface WebglFilterBackend extends FilterBackend, WebglFilterBackendOptions {
+	setupGLContext(width: number, height: number): void;
+
+	chooseFastestCopyGLTo2DMethod(width: number, height: number): void;
+
+	createWebGLCanvas(width: number, height: number): void;
+
+	applyFiltersDebug(filters: IBaseFilter[], sourceElement: HTMLImageElement | HTMLCanvasElement, sourceWidth: number, sourceHeight: number, targetCanvas: HTMLCanvasElement, cacheKey: any): any;
+
+	glErrorToString(context: any, errorCode: any): string;
+
+	createTexture(gl: WebGLRenderingContext, width: number, height: number, textureImageSource: HTMLImageElement | HTMLCanvasElement): WebGLTexture;
```

Comment:
`textureImageSource` seems to be [optional](https://github.com/kangax/fabric.js/blob/1fa3fdae31d2ee9a1c7c4bebb975d66b3171b5e9/src/filters/base_filter.class.js#L172)

Reply:
There's also another overload seen [here](https://github.com/kangax/fabric.js/blob/75ba9e1ac82a36fa28aed3660c90896ba864f017/src/filters/blendimage_filter.class.js#L107)